parkhangs-webapp/src/features/modal/ModalMapDetail.js

```diff
@@ -1,14 +1,42 @@
 import React from 'react'
 import AddEventForm from 'components/AddEventForm'
-import { deleteEvent } from 'features/parks/parksSlice.js'
+import { deleteEvent,fetchEventsById } from 'features/parks/parksSlice.js'
 import { connect } from 'react-redux'
 import 'features/modal/ModalMapDetail.css'
+import {unwrapResult} from '@reduxjs/toolkit'
+
 
 class ModalMapDetail extends React.Component {
+  
     selectedPark = () => {
         return this.props.parks.find(park => park.parkName === this.props.park)
     }
+   
+    componentDidMount = async () => {
+        let result =  await this.getEvents();
+        //console.log(result); 
+        //console.log(this.props.eventsById);
+        console.log(this.props.error);
+        if(this.props.error){
+            throw this.props.error
+        }
+        
+    }
 
+    getEvents = async () => {
+        try{
+             const resultAction = await this.props.fetchEvents(this.selectedPark().id) //waits for promise
+             const events = unwrapResult(resultAction); //resolving the promise and actually getting back the payload
```

Comment:
it might be a good idea to unwrap the result in the thunk and return the data from there instead of mixing redux logic into the component